authors/apps/authentication/tests/test_social_auth.py

```diff
@@ -0,0 +1,62 @@
+# Django imports
+from rest_framework.views import status
+
+# local imports
+from .base_test import TestBaseCase
+
+
+class TestSocialAuth(TestBaseCase):
+
+    def test_successful_oauth2_login(self):
+        """test successful oauth2 authentication with facebook"""
+
+        response = self.client.post(
+            self.social_auth_url, self.oauth2_data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.data["token"])
+
+    def test_successful_oauth1_login(self):
+        """test successful oauth1 authentication with twitter"""
+
+        response = self.client.post(
+            self.social_auth_url, self.oauth1_data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.data["token"])
+
+    def test_login_no_token(self):
+        """test login with token not provided"""
+        response = self.client.post(
+            self.social_auth_url, self.no_token_data, format='json')
+        self.assertTrue(response.data["errors"])
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_login_with_invalid_token(self):
+        """test login with a token that's invalid"""
+        response = self.client.post(
+            self.social_auth_url, self.invalid_token_data, format='json')
+        self.assertEqual(response.data["error"], "Invalid token")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_login_no_provider(self):
+        """test social login with provider name not provided"""
+        response = self.client.post(
+            self.social_auth_url, self.no_provider_data, format='json')
+        self.assertTrue(response.data["errors"])
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_login_no_access_token_secret(self):
+        """test social login with no access token secret provided"""
+        response = self.client.post(
+            self.social_auth_url, self.no_token_secret_data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
The 400 status code is repeated 4 times, maybe you could implement DRY

Reply:
I have worked on this